src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/IsolatedComponentLoadContext.cs

```diff
@@ -23,13 +24,21 @@ public IsolatedComponentLoadContext(string componentAssemblyPath) : base($"Isola
 
         protected override Assembly? Load(AssemblyName assemblyName)
         {
-            string? assemblyPath = _resolver.ResolveAssemblyToPath(assemblyName);
-            if (assemblyPath != null)
+            if (Marshal.IsComSupported)
```

Comment:
As we chatted about offline, `IsolatedComponentLoadContext` and `ComponentActivator` aren't actually tied to built-in COM support. (I can see the desire to be able to remove them for trimming / not having trim warnings, but lumping them under built-in COM support seems off)